core/DataAccess/ArchiveTableDao.php

```diff
@@ -60,7 +60,7 @@ public function getArchiveTableAnalysis($tableDate)
                        SUM(CASE WHEN name NOT LIKE 'done%' THEN 1 ELSE 0 END) AS count_numeric_rows,
                        0 AS count_blob_rows
                   FROM `$numericTable`
-              GROUP BY idsite, date1, date2, period";
```

Comment:
There is some mildly interesting difference between `idsite, date1, date2, period` and `idsite, period, date1, date2` for MySQL, probably because the index pre-sorts in one case:

```
# idsite, date1, date2, period
-> Group aggregate: [SELECT stuff]  (cost=828 rows=74) (actual time=0.472..20.2 rows=74 loops=1)
    -> Index scan on archive_numeric_2022_01 using index_idsite_dates_period  (cost=426 rows=4020) (actual time=0.188..12.6 rows=4431 loops=1)

# idsite, period, date1, date2
-> Sort: archive_numeric_2022_01.idsite, archive_numeric_2022_01.period, archive_numeric_2022_01.date1, archive_numeric_2022_01.date2  (actual time=21.8..21.9 rows=74 loops=1)
    -> Stream results  (cost=828 rows=74) (actual time=0.51..21.6 rows=74 loops=1)
        -> Group aggregate: [SELECT stuff]  (cost=828 rows=74) (actual time=0.488..21.2 rows=74 loops=1)
            -> Index scan on archive_numeric_2022_01 using index_idsite_dates_period  (cost=426 rows=4020) (actual time=0.192..12.5 rows=4431 loops=1)
```

Can that extra sorting stage create any problems? For example when having a five digit count of sites? Or would that be too extreme to reasonably consider?

Reply:
Hm. Really good question. From a user point of view, I thought it might be more useful to sort by period before sorting by date1. But there hadn't been any complaints about that before, so guess we could also go with a different sorting there.
Can change that if needed, but on the other side I wouldn't consider that as a big problem here, as the query is only performed in that specific command 🤔

Reply:
Sorting has no real measurable impact on an instance with 10k sites, and with that size the change in sorting (at least for me) looks more expected.

All other problems I could find are not related to any change done here.